### PR TITLE
Added API change to support prompting the user to login via MM

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
#673 

#### What's this PR do?
Backend changes to require users with MM-only social auth to login via that method.

#### How should this be manually tested?

You could try to login with a micromasters-only user and verify you see the expected response from the backend. Dealing with the UI piece in a separate PR.

### CODE REVIEW

**Comment by noisecapella** [comment]
> You could try to login with a micromasters-only user and verify you see the expected response from the backend.

I'm trying this but I'm seeing only the usual response. What should I look for?

**Comment by rhysyngsun** [comment]
I'm seeing this too, let me investigate.

**Comment by rhysyngsun** [comment]
@noisecapella I've investigated this and the error I was getting was this one:

https://github.com/python-social-auth/social-core/blob/master/social_core/pipeline/social_auth.py#L75

Which PSA raises when you have multiple users with the same email address, which really shouldn't be the case (I'm guessing with all the auth changes in the last month, maybe you got into a weird state too?). I'll check in prod too to be sure we won't run into it there either.

**Comment by noisecapella** [comment]
Waiting on the open-discussions-client fix

File: authentication/pipeline/user.py [hunk 1]
```diff
@@ -164,3 +168,32 @@ def validate_password(
...
+    """
+    # only do this check if the user is attempting an email login
+    if backend.name != EmailAuth.name or flow != SocialAuthState.FLOW_LOGIN or user is None:
+        return {}
+
+    user_storage = strategy.storage.user
+
+    # if the user only has one social auth and it is MicroMasters, prompt them to login via that method
+    if (
+            user_storage.get_social_auth_for_user(user).count() == 1 and
```
[the remark this answers is not in the document]
Comment (rhysyngsun) [reply authentication/pipeline/user.py]: Good catch!

[the remark this answers is not in the document]
Comment (rhysyngsun) [reply authentication/pipeline/user.py]: Actually, I just ended up reversing the query order, that'll basically have the same effect of addressing the race condition by asserting the count last, but avoid having to load all of the user's social auths just to assert that they only have one and it is Micromasters. The only way to break this is if someone goes into django-admin and changes the provider type between each query.

**Comment by rhysyngsun** [comment]
Ah, this has been approved. :joy:

**Pull request merged**